ignite/metrics/rouge.py

```diff
@@ -0,0 +1,151 @@
+from collections import defaultdict
+from typing import Callable, DefaultDict, Dict, List, Optional, Union
+
+import torch
+
+from ignite.exceptions import NotComputableError
+from ignite.metrics import Metric
+
+# These decorators helps with distributed settings
+from ignite.metrics.metric import reinit__is_reduced, sync_all_reduce
+
+
+class Rouge(Metric):
+    r"""Calculates the Rouge Score for two Sequence of Tokens.
+
+
+    .. math::
```

Comment:
I suppose Rouge-L is considered here. Formulas don't match the reference paper. Is it normal ?

Reply:
The one in the paper is only ROUGE- recall. Historically, ROUGE-recall and BLEU-precision were used, but later on ROUGE-precision and ROUGE-f1-score were introduced. Currently ROUGE-f1 average is the most widely used metric.
Also, I will add example and test cases for the metric in the next commit.

Reply:
Is a simple example like this one fine?
```
from ignite.metrics import Rouge
import torch
m = Rouge(alpha=1,n='l')
peer = "the cat was found under the bed"
model = "the cat was under the bed"
m.update([peer.split(),[model.split()]])
peer = "the tiny little cat was found under the big funny bed"
model = "the cat was under the bed"
m.update([peer.split(),[model.split()]])
print(m._num_examples)
print(m.compute())
```